### PR TITLE
cf-tables version 1.0.0 is ready!

### PR DESCRIPTION
**Pull request by mistergone**
This PR addresses the accessibility question for sortable tables, and cleans up a few minor things.
## Additions
- Styles for `button.sortable` which mostly just strips the default button styles (and makes the button full width and height to fill the TH it's in)
## Changes
- `.sortable` changed to a button to address accessibility
- JavaScript changes to match move to `.sortable` being a button
- Updated docs to note button preference for .sortable
- CSS properties re-ordered to better meet standards
## Testing
- Pull it down, check out the demo page or docs page.
- `grunt test`
## Review
- @Scotchester and @KimberlyMunoz and @jimmynotjim and anyone else who's interested!
## Todos
- I'd like to make tables sortable by date
- I'd like to expose public methods so that other apps can sort tables with those methods
## Checklist
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
- [x] Passes all existing automated tests
- [x] New functions include new tests
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged
- [x] Visually tested in supported browsers and devices 
- [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
## Special Thanks

Thanks to everyone for their help on this, including @Scotchester and @KimberlyMunoz and @jimmynotjim and @ascott1 and @anselmbradford and @sebworks!
## Animated GIF

It's a turn<strong>TABLE</strong>. Get it? Get it? :wink: :grin: :rocket: :octopus:

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 All notable changes to this project will be documented in this file.
 We follow the [Semantic Versioning 2.0.0](http://semver.org/) format.
 
+## 1.0.0 - 2015-11-18
+
+### Changed
+ - .sortable changed to a button to address accessibility
+ - Updated docs to note button preference for .sortable
+ - CSS properties re-ordered to better meet standards
```
Comment (Scotchester) [inline CHANGELOG.md]: Incredibly trivial, but if you want to be consistent with the other lists in this file, you could remove the spaces before the hyphens :)

File: src/cf-tables.less [hunk 2]
```diff
@@ -370,11 +380,24 @@
...
+        height: 100%;
+        padding: inherit;
+        margin: inherit;
+        border: none;
+        background: none;
+        font-family: inherit;
+        font-weight: inherit;
+        outline: none;
+        text-align: left;
+    }
```
Comment (Scotchester) [inline src/cf-tables.less:394]: Could this be simplified by mixing in the [`.btn__link` modifier](http://cfpb.github.io/cf-buttons/docs/#button-link) from cf-buttons?

File: src/cf-tables.less [hunk 3]
```diff
@@ -436,14 +459,14 @@
 
     .table__entry-header-on-small {
         & > tbody td:first-child {
+            padding-bottom: .75em;
+            margin-bottom: unit( 10px / @base-font-size-px, em);
             margin-top: 0;
+            border-bottom: 1px solid @table-border;
```
Comment (Scotchester) [inline src/cf-tables.less]: Border should come before margin, as we follow the box model from inside out (width/height > padding > border > margin).

Also, while fixing that, let's order multiple rules of the same type (`margin-bottom` and `margin-top` in this case) in the same order as the shorthand rule would (top, left, bottom, right).

File: src/cf-tables.less [hunk 4]
```diff
@@ -451,8 +474,8 @@
         }
 
         & > tbody > tr {
-            border-bottom: none;
             margin-bottom: unit( 30px / @base-font-size-px, em );
+            border-bottom: none;
```
Comment (Scotchester) [inline src/cf-tables.less:477]: As above, border before margin.

File: src/js/cf-tables.js [hunk 5]
```diff
@@ -18,7 +18,7 @@
     // rows is an Array of Arrays, serving as a model of the table
     rows = [],
     $table = $( table ),
-    $headercells = $table.find( '.sortable' );
+    $sortButton = $table.find( '.sortable' );
```
Comment (Scotchester) [inline src/js/cf-tables.js]: Should this be plural like the previous var name?

File: src/js/cf-tables.js [hunk 6]
```diff
@@ -104,30 +104,32 @@
...
-        if ( $cell.hasClass( 'sorted-up' ) || $cell.hasClass( 'sortable__start-down' ) ) {
+        if ( $button.hasClass( 'sorted-up' ) || $button.hasClass( 'sortable__start-down' ) ) {
           sign = -1;
         }
 
         // Remove existing sort classes in the table
-        $headercells.removeClass( 'sortable__start-down sortable__start-up' );
-        $headercells.removeClass( 'sorted-down sorted-up' );
+        $table.find( '.sortable' ).removeClass( 'sortable__start-down sortable__start-up' );
+        $table.find( '.sortable' ).removeClass( 'sorted-down sorted-up' );
```
Comment (Scotchester) [inline src/js/cf-tables.js]: Why not use `$sortButton` here in place of the jQuery find function, as you did with `$headercells` previously?

File: src/cf-tables.less (hunk 2 above)
Comment (mistergone) [inline src/cf-tables.less:394]: Yes! Except cf-tables doesn't currently require cf-buttons. Should we file this as an Issue that we fix once cf-tables makes it into the Voltrazord? Or is there a better fix?

File: src/cf-tables.less (hunk 3 above)
Comment (mistergone) [inline src/cf-tables.less]: Oops! Oversight. Fixed!

File: CHANGELOG.md (hunk 1 above)
Comment (mistergone) [inline CHANGELOG.md]: No problem. Fixed. :grin:

File: src/js/cf-tables.js (hunk 5 above)
Comment (mistergone) [inline src/js/cf-tables.js]: Good point, fixed!

File: src/js/cf-tables.js (hunk 6 above)
Comment (mistergone) [inline src/js/cf-tables.js]: Uh... I have no idea why I did that. Fixed!

File: src/cf-tables.less (hunk 2 above)
Comment (Scotchester) [inline src/cf-tables.less:394]: Yeah, it's a good point. I thought about that but decided not to mention it.

Here's an idea: What if we created a `.u-unstyled-button` utility class/mixin in cf-core and used that both here and as a base for `.btn__link`? I don't think I'd hold up this PR while waiting for that to happen, but tell me if you think that's a good idea.

/cc @ascott1 @jimmynotjim @KimberlyMunoz

Comment (mistergone) [inline src/cf-tables.less:394]: > Here's an idea: 

I like this idea! I had thought of something similar - afterall, `button` is extremely useful as an interactive element for apps, and including an easy way to have an unstyled `button` will come in handy for us in the future, I think.

Comment (Scotchester) [inline src/cf-tables.less:394]: Cool. Filed an issue (cfpb/capital-framework#245) to do that.

**Comment by Scotchester** [comment]
:+1: Great work!

**Pull request merged**